code/game/machinery/limbgrower.dm

```diff
@@ -0,0 +1,264 @@
+#define LIMBGROWER_MAIN_MENU       1
+#define LIMBGROWER_CATEGORY_MENU   2
+#define LIMBGROWER_CHEMICAL_MENU   3
+//use these for the menu system
+
+
+/obj/machinery/limbgrower
+	name = "limb grower"
+	desc = "It grows new limbs using Synthflesh."
+	icon = 'icons/obj/machines/limbgrower.dmi'
+	icon_state = "limbgrower_idleoff"
+	density = 1
+	flags = OPENCONTAINER
+
+	var/operating = 0
+	anchored = 1
+	use_power = 1
+	var/disabled = 0
+	idle_power_usage = 10
+	active_power_usage = 100
+	var/busy = 0
+	var/prod_coeff = 1
+
+	var/datum/design/being_built
+	var/datum/research/files
+	var/selected_category
+	var/screen = 1
+	var/emag = 0 //Gives access to other cool stuff
+
+	var/list/categories = list(
+							"human",
+							"lizard",
+							"plasmaman",
+							"special"
+							)
+
+	var/mob/living/carbon/human/human
+
+
+
+/obj/machinery/limbgrower/New()
+	..()
+	create_reagents(0)
+	var/obj/item/weapon/circuitboard/machine/B = new /obj/item/weapon/circuitboard/machine/limbgrower(null)
+	human = new /mob/living/carbon/human()
+	B.apply_default_parts(src)
+
+
+	reagents.add_reagent("synthflesh",50)
+
+	files = new /datum/research/limbgrower(src)
+
+/obj/item/weapon/circuitboard/machine/limbgrower
+	name = "circuit board (Limb Grower)"
+	build_path = /obj/machinery/limbgrower
+	origin_tech = "programming=2;biotech=2"
+	req_components = list(
+							/obj/item/weapon/stock_parts/manipulator = 1,
+							/obj/item/weapon/reagent_containers/glass/beaker = 2,
+							/obj/item/weapon/stock_parts/console_screen = 1)
+
+/obj/machinery/limbgrower/interact(mob/user)
+	if(!is_operational())
+		return
+
+	var/dat = main_win(user)
+
+	switch(screen)
+		if(LIMBGROWER_MAIN_MENU)
+			dat = main_win(user)
+		if(LIMBGROWER_CATEGORY_MENU)
+			dat = category_win(user,selected_category)
+		if(LIMBGROWER_CHEMICAL_MENU)
+			dat = chemical_win(user)
+
+	var/datum/browser/popup = new(user, "Limb Grower", name, 400, 500)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/limbgrower/deconstruction()
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		reagents.trans_to(G, G.reagents.maximum_volume)
+
+/obj/machinery/limbgrower/attackby(obj/item/O, mob/user, params)
+	if (busy)
+		user << "<span class=\"alert\">The Limb Grower is busy. Please wait for completion of previous operation.</span>"
+		return 1
```

Comment:
why do some returns return 1 and then others return null?

Should it not be 1 and 0, and if so, use the TRUE/FALSE defines